expo/expo
# [android] Remove old manifest and bundle loading code

# Why

This code is considered unused since preloading into the legacy cache and DB doesn't do anything (ExpoUpdatesAppLoader uses its own cache and DB). Because of this, we can delete quite a large chunk of legacy code.

We theoretically could replace this with ExpoUpdatesAppLoader but @esamelson mentioned that it probably wasn't worth it as this code is most likely a remnant of the time when apps didn't ship with an embedded manifest.

# How

1. Remove dead code, go to step 1.

# Test Plan

1. Build and run.
2. `adb shell am broadcast -a com.android.vending.INSTALL_REFERRER -n host.exp.exponent/.referrer.InstallReferrerReceiver --es "referrer" "hello=world"`
3. See the following in logcat: `D/InstallReferrerReceiver: Referrer: hello=world`
